packages/base/text-input-filter.gts

```diff
@@ -30,20 +32,21 @@ export class TextInputFilter<T> {
     private getValue: () => T | null,
     private setValue: (val: T | null | undefined) => void,
     private deserialize: (
-      value: string | null | undefined
+      inputValue: string | null | undefined
     ) => DeserializedResult<T>,
     private serialize: (val: T | null | undefined) => string | undefined = (
       v
     ) => (!v ? undefined : String(v))
   ) {}
-  @tracked result: DeserializedResult<string> | undefined;
+  @tracked _lastEditedInputValue: string | undefined;
+  @tracked _errorMessage: string | undefined;
 
   get asString(): string {
     let serialized = this.serialize(this.getValue());
-    if (serialized != null && this.result?.value !== serialized) {
+    if (serialized) {
```

Comment:
I think there may be something recursive going on. 

For example, if I put `this._lastEditedInputValue` in this code block
- type in gibberish
- my cursor moves to the end

I dont know a good explanation for this but it seems the InputTag component is rerendered. Since asString is just a visual representation of the users input, I dont see why we need to check that the serialized value is equals to the lastEditingInput here.

Reply:
I think this is the same behaviour we want 
- if json data is cannot be serialized, it wud display empty string (this is also for fields like big int which is more convincing since the serialize is custom)
- if during edit, the data cannot be serialized, it wud display the lastEditedInputValue and display errors

Reply:
@ef4 this undoes this change. Any feedback on this